blobfuse/include/BlobStreamer.h

```diff
@@ -0,0 +1,191 @@
+#ifndef BLOBSTREAMER_H_
+#define BLOBSTREAMER_H_
+
+#include <blobfuse.h>
+#include <BlobfuseGlobals.h>
+#include<StorageBfsClientBase.h>
+
+#include <FileLockMap.h>
+#include <list>
+
+using namespace std;
+using namespace azure::storage_lite;
+using namespace azure::storage_adls;
+
+class CacheSizeCalculator
+{
+    public:
+        static CacheSizeCalculator* GetObj();
+
+        void SetMaxSize(uint64_t max) {
+            max_usage = max;
+        }   
+
+        void AddSize(uint64_t bytes) {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            current_usage += bytes; 
+        }
+
+        void RemoveSize(uint64_t bytes) {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            if (current_usage < bytes)
+                current_usage = 0;
+            else 
+                current_usage -= bytes; 
+        }
+
+        uint64_t GetUsage() {
+            return current_usage;
+        }
+
+        bool MaxLimitReached() {
+            bool flag = current_usage >= max_usage;
+            if (flag) {
+                syslog(LOG_INFO, "Current block cache usage has reached the configured limit %lu > %lu", current_usage, max_usage);
+            }
+
+            return flag;
+        }
+
+    private:
+
+        CacheSizeCalculator() {
+            current_usage = max_usage = 0;
+        }
+
+        static CacheSizeCalculator* mInstance;  // Static object for singleton class
+
+        uint64_t        current_usage;          //  Current memory usage based on how many blocks are cached
+        uint64_t        max_usage;              //  Max memory usage allowed by configuration for caching
+        std::mutex      m_mutex;                //  Mutex to sync the map
+
+};
+
+// Structure representing one block segment
+struct BlobBlock {
+        bool                valid;  // Block is valid or not
+        bool                last;   // This block is the last block of file
+        uint64_t            start;  // Start offset of block
+        uint64_t            length;    // End offset of block
```

Comment:
Is this the end offset of the block or is that start + length?

Reply:
length of the data inside block. Using length i can count endindex so keeping start offset and length of data as metadata.

Reply:
Renamed to start_offset and data_length. Dont want to use index as for file offset makes logical sense compared to index.